File: Back/SoftRhUsuariosBO/src/pe/edu/pucp/softrh/usuarios/bo/UsuarioBO.java
package pe.edu.pucp.softrh.usuarios.bo;

import java.util.ArrayList;
import pe.edu.pucp.softrh.usuarios.dao.UsuarioDAO;
import pe.edu.pucp.softrh.usuarios.daoimp.UsuarioDAOImp;
import pe.edu.pucp.softrh.usuarios.model.Usuario;

public class UsuarioBO {
	private UsuarioDAO usuarioDAO;

	public UsuarioBO() {
		this.usuarioDAO = new UsuarioDAOImp();
	}

	public Integer insertar(Usuario usuario) {
		return usuarioDAO.insertar(usuario);
	}

	public Integer modificar(Integer idUsuario, String nombres, String apellidos) {
		return usuarioDAO.modificar(idUsuario, nombres, apellidos);
	}

	public Integer eliminar(Integer idUsuario) {
		return usuarioDAO.eliminar(idUsuario);
	}

	public ArrayList<Usuario> listarTodos() {
		return usuarioDAO.listarTodos();
	}

	public Usuario obtenerPorId(Integer idUsuario) {
		return usuarioDAO.obtenerPorId(idUsuario);
	}

	public Integer verificarIngresoUsuario(String correo, String contrasenha) {
		return usuarioDAO.verificarIngresoUsuario(correo, contrasenha);
	}

	public String obtenerRolUsuario(String correo, String contrasenha) {
		return usuarioDAO.obtenerRolUsuario(correo, contrasenha);
	}

	public Integer verificarContrasenha(Integer idUsuario, String contrasenha) {
		return usuarioDAO.verificarContrasenha(idUsuario, contrasenha);
	}

	public Integer cambiarContrasenha(Integer idUsuario, String contrasenhaNueva) {
		return usuarioDAO.cambiarContrasenha(idUsuario, contrasenhaNueva);
	}

	public Integer resetearContrasenha(Integer idUsuario) {
		return usuarioDAO.resetearContrasenha(idUsuario);
	}
}

File: Front/RHStoreWS/RHStoreUsuariosBO/UsuarioBO.cs
using RHStoreBaseBO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RHStoreUsuariosBO
{
	public class UsuarioBO : BaseBO
	{
		public int modificarUsuario(int idUsuario, string nombres, string apellidos)
		{
			return UsuarioWS.modificarUsuario(idUsuario, nombres, apellidos);
		}

        public int verificarIngresoUsuario(string correo, string contrasenha)
        {
            return UsuarioWS.verificarIngresoUsuario(correo, contrasenha);
        }

		public string obtenerRolUsuario(string correo, string contrasenha)
		{
			return UsuarioWS.obtenerRolUsuario(correo, contrasenha);
		}

		public int verificarContrasenha(int idUsuario, string contrasenha)
		{
			return UsuarioWS.verificarContrasenha(idUsuario, contrasenha);
		}

		public int cambiarContrasenha(int idUsuario, string contrasenhaNueva)
		{
			return UsuarioWS.cambiarContrasenha(idUsuario, contrasenhaNueva);
		}

		public int resetearContrasenha(int idUsuario)
		{
			return UsuarioWS.resetearContrasenha(idUsuario);
		}
	}
}
